src/components/Pressable.tsx

```diff
@@ -0,0 +1,277 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { GestureObjects as Gesture } from '../handlers/gestures/gestureObjects';
+import { GestureDetector } from '../handlers/gestures/GestureDetector';
+import {
+  GestureStateChangeEvent,
+  GestureTouchEvent,
+  TouchData,
+} from '../handlers/gestureHandlerCommon';
+import { PressEvent, PressableEvent, PressableProps } from './PressableProps';
+import { HoverGestureHandlerEventPayload } from '../handlers/gestures/hoverGesture';
+import { LongPressGestureHandlerEventPayload } from '../handlers/LongPressGestureHandler';
+import { Insets, StyleProp, View, ViewStyle } from 'react-native';
+import RNButton from './GestureHandlerButton';
+
+const DEFAULT_LONG_PRESS_DURATION = 500;
+const DEFAULT_HOVER_DELAY = 0;
+
+const numberAsInset = (value: number): Insets => ({
+  left: value,
+  right: value,
+  top: value,
+  bottom: value,
+});
+
+const addInsets = (a: Insets, b: Insets): Insets => ({
+  left: (a.left ?? 0) + (b.left ?? 0),
+  right: (a.right ?? 0) + (b.right ?? 0),
+  top: (a.top ?? 0) + (b.top ?? 0),
+  bottom: (a.bottom ?? 0) + (b.bottom ?? 0),
+});
+
+const touchToPressEvent = (data: TouchData, timestamp: number): PressEvent => ({
+  identifier: data.id,
+  locationX: data.x,
+  locationY: data.y,
+  pageX: data.absoluteX,
+  pageY: data.absoluteY,
+  target: 0, // fixme if possible, set to correct target ID
```

Comment:
What is the correct target ID in this case?

Reply:
That'd be the `RNButton`'s `Native ID`. I know it's there i just have to find a way to access it.

Reply:
I'm almost sure events are not bubbling. I think target is present in the events on iOS, not sure if it's present on other platforms so it may not be useful.

If you cannot figure out a way to get it, it may make sense to implement it on Android & Web.